### PR TITLE
Sane SSZ object default values

### PR DESCRIPTION
**Pull request by JustinDrake**
Remove 92 lines in phase 0, along with a bunch of useless constants and helpers 🎉

### CODE REVIEW

File: specs/core/0_beacon-chain.md [hunk 1]
```diff
@@ -1373,35 +1352,7 @@ When enough full deposits have been made to the deposit contract, an `Eth2Genesi
...
-* Let `genesis_block = get_empty_block()`.
-* Set `genesis_block.state_root = hash_tree_root(genesis_state)`.
-
-```python
-def get_empty_block() -> BeaconBlock:
-    """
-    Get an empty ``BeaconBlock``.
-    """
-    return BeaconBlock(
-        slot=GENESIS_SLOT,
```
[the remark this answers is not in the document]
Comment (JustinDrake) [reply specs/core/0_beacon-chain.md]: `GENESIS_SLOT` can probably go away completely now

[the remark this answers is not in the document]
Comment (djrtwo) [reply specs/core/0_beacon-chain.md]: Would there never be a reason to specify GENESIS_SLOT as greater than 0?

`GENESIS_SLOT` is used as a comparator in many locations and I think it is cleaner than `== 0`. I'd say keep `GENESIS_SLOT` around in the spec.

[the remark this answers is not in the document]
Comment (JustinDrake) [reply specs/core/0_beacon-chain.md]: > `GENESIS_SLOT` is used as a comparator in many locations 

There are only 5 matches of `GENESIS_SLOT` in the phase 0 doc now:

* The definition under "Initial values"
* `if state.slot == GENESIS_SLOT:` in `activate_validator` (this used to be `is_genesis`)
* At every `slot > GENESIS_SLOT` run the following function
* The steps below happen when `state.slot > GENESIS_SLOT`
* At every `slot > GENESIS_SLOT` run the following function

The last three can likely be combined.

[the remark this answers is not in the document]
Comment (JustinDrake) [reply specs/core/0_beacon-chain.md]: I'll keep `GENESIS_SLOT` and `GENESIS_EPOCH` and try to minimise/reduce the use of these constants in another PR. My gut feel is that we don't want to encourage people modifying `GENESIS_SLOT`.

**Pull request merged**